scripts/exact-example.jl

```diff
@@ -0,0 +1,240 @@
+# important to set the `nev` and `which` keyowrd arguments. `nev` sets the number of
+# eigenpairs to find. `which` should in most cases be set to `:SR`, which will find the
+# eigenvalues with the smallest real part.
+
+using Arpack
+
+num_eigvals = 3
+
+sm = sparse(ham)
+vals_ar, vecs_ar = eigs(sm; which=:SR, nev=num_eigvals)
+vals_ar
+
+# Using KrylovKit's
+# [`eigsolve`](https://jutho.github.io/KrylovKit.jl/stable/man/eig/#KrylovKit.eigsolve) is
+# similar, but the `nev` and `which` are given as positional arguments. Note that KrylovKit
+# may sometimes return more than `nev` eigenpairs if it happens to find them.
+
+using KrylovKit
+
+vals_kk, vecs_kk = eigsolve(sm, num_eigvals, :SR)
+vals_kk
+
+# Both solvers use variants of the [Lanczos
+# algorithm](https://en.wikipedia.org/wiki/Lanczos_algorithm) for Hermitian matrices and the
+# [Arnoldi algorithm](https://en.wikipedia.org/wiki/Arnoldi_iteration) for non-Hermitian
+# ones. These may in some cases miss degenerate eigenpairs.
+
+# If diagonalisation takes too long, you can reduce the tolerance by setting the `tol`
+# keyword argument to `eigs` or `eigsolve`. Using drastically lower tolerances than the
+# default can still produce good results in practice. This, however, should be checked on a
+# case-by-case basis.
+
+# ### The matrix-free method
+
+# KrylovKit's
+# [`eigsolve`](https://jutho.github.io/KrylovKit.jl/stable/man/eig/#KrylovKit.eigsolve)
+# function is implemented in a way that does not require the linear operator and vector to
+# be Julia arrays. Rimu leverages this functionality, which allows diagonalising
+# Hamiltonians without ever needing to construct the matrix - all matrix elements are
+# generated on the fly.
+
+# While this method is by far the slowest of the ones discussed, it also uses drastically
+# less memory. This allows us to diagonalise much larger Hamiltonians.
+
+# To use this method, you first need a starting [`DVec`](@ref):
+
+dvec = DVec(add => 1.0)
+
+# Then, pass that vector and the Hamiltonian to
+# [`eigsolve`](https://jutho.github.io/KrylovKit.jl/stable/man/eig/#KrylovKit.eigsolve). Since
+# the function has no way of knowing the Hamiltonian is Hermitian, we have to provide that
+# information through the `issymmetric` or `ishermitian` keyword arguments. Make sure to
+# only pass this argument when the Hamiltonian is actually symmetric. To check that,
+# look at the `LOStructure` trait:
+
+LOStructure(ham)
+
+#
+
+vals_mf, vecs_mf = eigsolve(ham, dvec, num_eigvals, :SR; issymmetric=true)
+vals_mf
+
+# Keep in mind that if an eigenvector is orthogonal to `dvec`, KrylovKit will miss
+# it. Consider the following example:
+
+eigsolve(ham, vecs_mf[2], num_eigvals, :SR, issymmetric=true)[1]
+
+# ## Reducing matrix size with symmetries
+
+# As these matrices tend to get large quickly, memory is usually the bottleneck.  There are
+# currently two methods implemented to reduce the matrix size, [`ParitySymmetry`](@ref) and
```

Comment:
I think this should briefly mention why this works/what is happening. That is, without explaining exactly what these symmetries mean, at least say that you are reducing the full matrix by selecting one of the blocks of block diagonal form.